noxfile.py

```
@@ -152,6 +155,72 @@ def lint(session: nox.Session) -> None:
+    # make sure we have a dist dir for pyodide
+    dist_dir = None
+    if "PYODIDE_ROOT" in os.environ:
+        # we have a pyodide build tree checked out
+        # use the dist directory from that
+        dist_dir = Path(os.environ["PYODIDE_ROOT"])
+    else:
+        # we don't have a build tree, get one
+        # that matches the version of pyodide build
+        import pyodide_build  # type: ignore[import]
+
+        pyodide_version = pyodide_build.__version__
+        pyodide_artifacts_path = Path(session.cache_dir) / f"pyodide-{pyodide_version}"
+        if not pyodide_artifacts_path.exists():
+            print("Fetching pyodide build artifacts")
+            session.run(
+                "wget",
+                f"https://github.com/pyodide/pyodide/releases/download/{pyodide_version}/pyodide-{pyodide_version}.tar.bz2",
+                "-O",
+                f"{pyodide_artifacts_path}.tar.bz2",
+            )
+            pyodide_artifacts_path.mkdir(parents=True)
+            session.run(
+                "tar",
+                "-xjf",
+                f"{pyodide_artifacts_path}.tar.bz2",
+                "-C",
+                str(pyodide_artifacts_path),
+                "--strip-components",
+                "1",
+            )
+
+        dist_dir = pyodide_artifacts_path
+    assert dist_dir is not None
+    assert dist_dir.exists()
+    if runner == "chrome":
+        # install chrome webdriver and add it to path
+        from webdriver_manager.chrome import ChromeDriverManager  # type: ignore[import]
+
+        driver = ChromeDriverManager().install()
+        session.env["PATH"] = f"{Path(driver).parent}:{session.env['PATH']}"
+
+        tests_impl(
+            session,
+            "emscripten-test",
+            pytest_extra_args=[
+                "--rt",
+                "chrome-no-host",
+                "--dist-dir",
+                str(dist_dir),
+                "test",
+            ],
+        )
```

We should fail if we receive an unknown runner (ie `else: raise ValueError(...)`).